src/gen.ml

```diff
@@ -86,6 +86,18 @@ let substitute_vars s vars subst_param =
           parami + 1
       in
       loop acc i2 parami tl
+    | SingleIn param :: tl ->
```

Comment:
it seems SingleIn semantics more precisely is Single param with text substitution (as opposed prepared statements parameter for Single), so lets rename to say `SingleText`?